src/uri/port.rs

```diff
@@ -46,41 +60,58 @@ impl<'a> Port<'a> {
     /// let port = authority.port_part().unwrap();
     /// assert_eq!(port.as_str(), "80");
     /// ```
-    pub fn as_str(&self) -> &'a str {
-        self.bytes
+    pub fn as_str(&self) -> &str {
+        self.repr.as_ref()
+    }
+}
+
+impl<T> fmt::Debug for Port<T> {
```

Comment:
For forwards compat, do we want to bound `T: fmt::Debug` even if it is not included?